### DISCUSSION TITLE
Export/import MSIX apps

### DISCUSSION THREAD

**Discussion by Filisteos** [Micke-K/IntuneManagement#191] [General]
Hello!
I have tried to export the new Microsoft Teams (MSIX format) to later import it into another tenant.
The first thing I have noticed is that during the export, the file is not exported. I thought there might be problems as there are with win32 so I manually put the .msix in the same folder as the exported .json. When trying to import what I notice is that the app uploads too quickly, and in fact it is because it does not upload the file, the application in Intune is blocked waiting for the file to be uploaded that never finishes uploading. Is there a problem with MSIX? Is it a feature that has not yet been implemented in the app? Thank you so much!

**Comment by Micke-K** [comment]
Hello!

Not sure if I've tried to import and msix app. I'll have a look. Might not be until the weekend. 

Cheers!

**Reply by Filisteos** [reply]
Thank you so much. How could I make you a donation? Your software is very useful to me.

**Comment by Micke-K** [comment]
Hello!

Can you try the attached files?
**Note:** I have NOT tested an app installation, only creating the package and upload of MSIX file. Please let me know if you test it.
The test I did:

- Created a MSIX package in Intune
- Exported the app
- Copied the MSIX file to the export folder
- Imported the app with Always import.
- App created and msix file uploaded.

Export will have the same issues as intunewin files. They are encrypted during upload. 
You can enable "Save Encryption File" in Settings. You should then be able to download the app again if the json file with encryption info is available.

Sorry, no donation available :). I might add a hidden coffee button in there one day. 

Cheers!

[IntuneAppManagement.zip](https://github.com/Micke-K/IntuneManagement/files/14150518/IntuneAppManagement.zip)

**Reply by Filisteos** [reply]
Hello Micke, I have repeated the process with the latest version 3.9.5. Again the app never finishes uploading
![importFolder](https://github.com/Micke-K/IntuneManagement/assets/143745549/c802db50-2fe6-4a64-8818-6ed3d6a479dd)
<img width="953" alt="intuneAppMSIX" src="https://github.com/Micke-K/IntuneManagement/assets/143745549/6532c375-131a-4e9f-a0a6-65756a052155">

**Reply by Filisteos** [reply]
The .json file looks correct: 

    "displayName":  "Microsoft Teams (work or school)",
    "description":  "Microsoft Corporation",
    "publisher":  "Microsoft",
    "createdDateTime":  "2024-01-30T18:58:34.831789Z",
    "lastModifiedDateTime":  "2024-01-30T19:18:32.0524396Z",
    "isFeatured":  false,
    "privacyInformationUrl":  null,
    "informationUrl":  null,
    "owner":  "",
    "developer":  "",
    "notes":  "",
    "uploadState":  1,
    "publishingState":  "published",
    "isAssigned":  false,
    "roleScopeTagIds":  [
                            "0"
                        ],
    "dependentAppCount":  0,
    "supersedingAppCount":  0,
    "supersededAppCount":  0,
    "committedContentVersion":  "1",
    "fileName":  "MSTEAMS-X64.MSIX",
    "size":  129612864,
    "applicableArchitectures":  "x64",
    "applicableDeviceTypes":  "desktop",
    "identityName":  "MSTeams",
    "identityPublisherHash":  "8wekyb3d8bbwe",
    "identityResourceIdentifier":  "",
    "isBundle":  false,
    "identityVersion":  "23320.3021.2567.4799",

**Comment by Micke-K** [comment]
Hello,

Did you download the ZIP I attached earlier and replaced the two files from the zip?

The 3.9.5 version does not fix it. Only the two modules in the zip should fix it.

Cheers!

**Reply by Filisteos** [reply]
Sorry @Micke-K, I had not read it correctly, tomorrow I will try exactly as you indicate.

**Reply by Filisteos** [reply]
@Micke-K !! With the new 2 modules it is working correctly, the application is uploaded. Thank you so much!!
<img width="328" alt="Captura de pantalla 2024-02-09 141305" src="https://github.com/Micke-K/IntuneManagement/assets/143745549/3cec04b7-678c-4383-82f4-3af0e2b319c5">
<img width="536" alt="Captura de pantalla 2024-02-09 140451" src="https://github.com/Micke-K/IntuneManagement/assets/143745549/3ba508a3-859b-4850-b490-53654bd60a52">

**Reply by Filisteos** [reply]
Add a coffe button soon!!
